### PR TITLE
Do not use deprecated methods in documentation

### PR DESCRIPTION
**Pull request by kishida**
Not to use `respond` method.
Use lambda if it can.
~~Use `:name` style instead of `{name}` style.~~
Use `:name` style to indicate that Armeria allows it.
Fix the lack of braces.
https://github.com/line/armeria/issues/1041

Including some of https://github.com/line/armeria/pull/999
Should I include `import`?

### CODE REVIEW

File: site/src/sphinx/server-basics.rst [hunk 1]
```diff
@@ -106,7 +96,7 @@ Even if we opened a port, it's of no use if we didn't bind any services to them.
     sb.annotatedService(new Object() {
         @Get("/greet5")
         public HttpResponse greet(@Param("name") String name,
-                                  @Param("title") @Optional("Mr.") String title) {
+                                  @Param("title") /*@Optional("Mr.")*/ String title) {
```
Comment (kishida) [inline site/src/sphinx/server-basics.rst]: Where is the `@Optional`?

Comment (trustin) [reply site/src/sphinx/server-basics.rst]: It is now `@Default`.

**Comment by trustin** [comment]
> Should I include import?

Only when a class appears first time in the page. No need to import JDK classes.

**Comment by trustin** [comment]
Thanks a lot for fixing the documentation, @kishida!

**Comment by kishida** [comment]
Update～

File: site/src/sphinx/server-basics.rst [hunk 2]
```diff
@@ -45,53 +45,56 @@ Even if we opened a port, it's of no use if we didn't bind any services to them.
 
 .. code-block:: java
 
+    import com.linecorp.armeria.server.ServiceRequestContext;
+    import com.linecorp.armeria.server.AbstractHttpService;
```
Comment (trustin) [inline site/src/sphinx/server-basics.rst]: Could you sort the imports?

Comment (kishida) [reply site/src/sphinx/server-basics.rst]: Fixed 😃

**Comment by hyangtack** [comment]
@kishida Oops! I recently added [server-annotated-service.rst](https://github.com/line/armeria/blob/master/site/src/sphinx/server-annotated-service.rst) and it has been merged. I used `{name}` style path variables In that document. Do we better change them to `:name` style? WDYT?

**Comment by kishida** [comment]
@hyangtack I've changed to introduce the both style. 😄

**Comment by kishida** [comment]
Talking with Trustin-san on Slack, `{name}` style may be better since JAX-RS (and Spring MVC) are using it.

**Comment by trustin** [comment]
@kishida Thanks a lot!

File: site/src/sphinx/server-basics.rst [hunk 3]
```diff
@@ -45,53 +45,54 @@ Even if we opened a port, it's of no use if we didn't bind any services to them.
...
-            res.respond(HttpStatus.OK, MediaType.PLAIN_TEXT_UTF_8, "Hello, %s!", name);
+            return HttpResponse.of(HttpStatus.OK, MediaType.PLAIN_TEXT_UTF_8, "Hello, %s!", name);
         }
-    }.decorate(LoggingService.newDecorator()); // Enable logging
+    }.decorate(LoggingService.newDecorator())); // Enable logging
 
     // Using an annotated service object:
     sb.annotatedService(new Object() {
-        @Get("/greet2/{name}")
+        @Get("/greet2/:name") // `:name` style is also available
```
Comment (minwoox) [inline site/src/sphinx/server-basics.rst]: Didn't change it back to `{name}` :)

Comment (kishida) [reply site/src/sphinx/server-basics.rst]: Only here, I want to indicate that Armeria allows `:name` style.

**Comment by trustin** [comment]
Thanks a lot, @kishida !

**Comment by kishida** [comment]
I'm very happy to be able to commit Armeria!

**Pull request merged**